### PR TITLE
find UW star files avoiding hard-coded names

### PR DESCRIPTION
**Pull request by jchiang87**
This samples the desired healpixel, computing the corresponding HTM index at each point, and adds that corresponding file to the list.

### CODE REVIEW

File: skycatalogs/utils/star_parquet_input.py [hunk 1]
```diff
@@ -1,33 +1,47 @@
...
+        self.htm_indexer = HtmIndexer(depth=20)
 
-    fnames = ['stars_chunk_8800000000000_8900000000000.parquet',
-              'stars_chunk_9200000000000_9300000000000.parquet',
-              'stars_chunk_9700000000000_9800000000000.parquet']
+    def _index_files(self, input_dir):
+        if self._files:
+            # The files have already been indexed.
+            return
+        files = sorted(glob.glob(os.path.join(input_dir,
```
Comment (JoanneBogart) [inline skycatalogs/utils/star_parquet_input.py:22]: A description of the format for input file basenames would be helpful, either in a comment or use a regular expression to pick out fields of interest, which can be named, e.g.
`pat = "stars_chunk_(?P<min_htm_id>\d+)_(?P<max_htm_id>\d+).parquet"`
Then lines 25-28 below get replaced with
`m = re.match(pat, os.path.basename(item))`
`self._files[(m['min_htm_id'], m['max_htm_id'])] = item

File: skycatalogs/utils/star_parquet_input.py [hunk 2]
```diff
@@ -1,33 +1,47 @@
...
+                                              'stars_chunk_*.parquet')))
+        for item in files:
+            tokens = os.path.basename(item).split('_')
+            imin = int(tokens[2])
+            imax = int(tokens[3].split('.')[0])
+            self._files[(imin, imax)] = item
 
-    return [os.path.join(dirpath, fname) for fname in fnames]
+    def find_files(self, pixel, nside=32, res_factor=16):
```
Comment (JoanneBogart) [inline skycatalogs/utils/star_parquet_input.py:30]: A docstring for this routine would be welcome.  Why was 16 chosen as default for `res_factor`?

**Review by JoanneBogart — APPROVED** [review]
This looks ok.  There are a couple places noted in inline comments which could use a little more explanation.

File: skycatalogs/utils/star_parquet_input.py (hunk 1 above)
Comment (jchiang87) [reply skycatalogs/utils/star_parquet_input.py:22]: I dislike regular expressions, but if u really think this is better and generally easier for people to understand (I don't particularly), I can change it.

Comment (JoanneBogart) [reply skycatalogs/utils/star_parquet_input.py:22]: I find the regular expression clearer; I don't know whether most people do or not. Adding a comment describing the structure of the basename instead is fine with me.

File: skycatalogs/utils/star_parquet_input.py (hunk 2 above)
Comment (JoanneBogart) [reply skycatalogs/utils/star_parquet_input.py:30]: What does resfactor have to do with HTM depth (apparently 20), if anything? 
Healpixels with nside=32 are much bigger than HTM depth 20 regions (or depth 16 regions, for that matter). How did you verify that nside*16 gives a sufficiently fine grid?

Comment (JoanneBogart) [reply skycatalogs/utils/star_parquet_input.py:30]: Another way to do this (which seems safer to me) would be to define a minimum circle around the healpixel, then call HtmIndexer.getShardIds

**Pull request merged**